src/feature/articles/getSingleArticle/GetSingleArticleComponent.js

```diff
@@ -14,42 +18,78 @@ import bookmark from '../../../app/common/images/bookmark.png';
 import ShareArticle from '../shareArticle/ShareArticleComponent';
 import FollowUnfollowComponent from '../../followUnfollow/FollowUnfollowComponent';
 import CommentCountComponent from '../../../app/common/CommentCount/CommentCountComponent';
+import ArticleMenuDropdown from '../../../app/common/articleMenu/ArticleDropdownMenu';
+import deleteArticle from '../deleteArticle/DeleteAction';
 import './GetSingleArticle.scss';
 
 export class ViewSingleArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayMenu: false
+    };
+  }
+
   componentDidMount() {
-    const { GetSingleArticle } = this.props;
-    const { slug } = this.props.match.params;
+    const {
+      match: {
+        params: { slug }
+      },
+      GetSingleArticle
+    } = this.props;
     GetSingleArticle(slug);
     window.scrollTo(0, 0);
   }
 
+  componentDidUpdate() {
+    const { deleted } = this.props;
+    if (deleted) {
+      window.location.assign('/');
+    }
+  }
+
+  closeMenu = () => {
+    this.setState({ displayMenu: false }, () => {
+      document.removeEventListener('click', this.closeMenu);
+    });
+  };
+
+  displayMenu = () => {
+    this.setState({ displayMenu: true }, () => {
+      document.addEventListener('click', this.closeMenu);
+    });
+  };
+
   render() {
     const {
       article: {
-        author = {},
-        id,
-        authorId,
-        title,
-        description,
-        body,
-        slug,
-        likes,
-        dislikes,
-        createdAt,
-        readTime,
-        averageRatings,
-        commentCount
-      }
-    } = this.props.article;
+        article: {
```

Comment:
@Jkadhuwa any reason why this article object contains again only another article object ?

Reply:
This is because when I am mapping the state to props I used article to represent getSingleArticle state, inside the getSingleArticle  there is article again. That why I nested them.

Reply:
I was nesting because I mapped the getSingleArticle state to article, but I have resolved it.